spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ConstantColumnVector.java

```diff
@@ -119,6 +122,8 @@ public UTF8String getUTF8String(int rowId) {
 
   @Override
   public ColumnVector getChild(int ordinal) {
-    throw new UnsupportedOperationException("ConstantColumnVector only supports primitives");
+    DataType sparkType = ((StructType) type).fields()[ordinal].dataType();
+    Type childType = SparkSchemaUtil.convert(sparkType);
+    return new ConstantColumnVector(childType, batchSize, ((InternalRow) constant).get(ordinal, sparkType));
```

Comment:
It looks like the approach in this PR is to use existing constant handling to fix nested column construction. I'm not sure that this is the right approach. There isn't anything specific to ORC here, so I'm not convinced that the ORC readers handle nested constant values correctly. Avoiding a problem is an okay solution as long as we're confident that it is always avoided.

At a minimum, I think we should have tests for partitioning by nested fields in structs that are not constants and would not get created by `findMoreCompoundConstants`.

Reply:
> There isn't anything specific to ORC here,so I'm not convinced that the ORC readers handle nested constant values correctly

Ans: The change itself is not specific to ORC here, but ORC is currently used that way, and Parquet's vectorization read do not use this, and this is a general practice, not to ORC itself.

>  I'm not sure that this is the right approach.

Ans:  When doing the vectorization read for orc, ```VectorizedSparkOrcReaders#StructConverter```use ```idToConstant``` and will convert constant values to ```ConstantColumnVector```, but```ConstantColumnVector``` cannot represent a Struct Type currently. So I need a struct constant.

According to Spark's API, a ```ColumnVector``` that represents a struct must implement the ```getChild``` method.
https://github.com/apache/spark/blob/v3.2.0/sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ColumnVector.java#L219
https://github.com/apache/spark/blob/v3.2.0/sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ColumnVector.java#L302

Since ```getChild``` returns a ```ColumnVector```, and each child in a struct type is a constant also, so it can return a ```ConstantColumnVector``` I think.

Besides , I will add more tests.